codegen/src/main.rs

```diff
@@ -0,0 +1,50 @@
+mod codegen;
+mod cubemx;
+
+use anyhow::Result;
+use clap::{App, AppSettings, Arg, ArgMatches, SubCommand};
+use cubemx::Db;
+
+fn parse_args() -> ArgMatches<'static> {
+    App::new("codegen")
+        .about("Code generation for the stm32f3xx-hal crate")
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .subcommand(
+            SubCommand::with_name("gpio")
+                .about("Generate GPIO mappings from an STM32CubeMX database")
+                .arg(
+                    Arg::with_name("db_path")
+                        .help("Path of the STM32CubeMX MCU database (.../db/mcs/)")
+                        .required(true),
+                ),
+        )
+        .get_matches()
+}
```

Comment:
It might be work it to just use [`structopt`](https://docs.rs/structopt/0.3.18/structopt/)

Reply:
You're right, that makes the code a bit nicer :)